rewrite-pattern.js

```diff
@@ -518,10 +525,9 @@ const processCharacterClass = (
 						);
 					} else {
 						// Generate negative set directly when case folding is not involved.
-						update(
-							characterClassItem,
-							UNICODE_SET.clone().remove(singleChars).toString(regenerateOptions)
-						);
+						const negativeSet = UNICODE_SET.clone().remove(singleChars);
+						const bmpOnly = regenerateContainsAstral(negativeSet);
+						update(characterClassItem, negativeSet.toString({ bmpOnly: bmpOnly }));
```

Comment:
If the regenerate set spans from code points before surrogate to astral sets, `toString({ bmpOnly: false })` returns much more verbose results while `toString({ bmpOnly: false })` is already correct: I think it should be fixed in regenerate later.

```js
const regenerate = require('regenerate');
const set = regenerate().addRange(0xd000, 0x10000);

console.log(set.toString());
// [\uD000-\uD7FF\uE000-\uFFFF]|\uD800\uDC00|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF]


console.log(set.toString({ bmpOnly: true }));
// [\uD000-\uFFFF]|\uD800\uDC00
```

The latter is apparently correct as it matches lone surrogates as well as U+10000. The former seems like `[\uD000-\uFFFF]|\uD800\uDC00` is passed to the bmp pass again.